tests/rbd/rbd_trash_purge.py

```diff
@@ -0,0 +1,148 @@
+from ceph.parallel import parallel
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def create_snap_and_clone(rbd, pool, image_name, snap_name, clone_name):
+    """
+    Create snap, protect and create clone for the given image
+    Args:
+        rbd: rbd object
+        pool: pool name
+        image_name: image name
+        snap_name: snap name
+        clone_name: clone name
+    """
+    if rbd.snap_create(pool, image_name, snap_name):
+        log.error(f"Snap creation failed for {image_name}")
+        return 1
+    snap_spec = f"{pool}/{image_name}@{snap_name}"
+    if rbd.protect_snapshot(snap_spec):
+        log.error(f"Snap protect failed for {snap_spec}")
+        return 1
+    if rbd.create_clone(snap_spec, pool, clone_name):
+        log.error(f"Clone creation failed for {clone_name}")
+        return 1
+    return 0
+
+
+def rbd_trash_purge(rbd, pool_type, **kw):
+    """
+    Run trash and force remove operation on images
+    Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: Test data
+    """
+    pool = kw["config"][pool_type]["pool"]
+    num_images = kw["config"].get("no_of_images", 10)
+    num_clones = kw["config"].get("no_of_clones", 4)
+    client = kw["ceph_cluster"].get_nodes(role="client")[0]
+    images_with_clone = list()
+    try:
+        log.info(f"Creating {num_images} number of images in pool {pool}")
+
+        for i in range(0, num_images):
+            image_name = f"image_{i}"
+            if rbd.create_image(pool, image_name, "10G"):
+                log.error(f"Image creation failed for {image_name}")
+                return 1
+            run_fio(
+                image_name=image_name, pool_name=pool, client_node=client, runtime=10
+            )
```

Comment:
IMO we could move it to below parallel block

Reply:
I tried doing that. But multiple ssh connections for running fio was causing some error to me. Not sure if it's just a one time error or if it would continue to happen in pipeline. So I didn't wanna take the risk and I moved it back to serial execution itself.